### ISSUE TITLE
Applying Different Layouts to Different Sites

### ISSUE THREAD

**Issue by remesq** [PiranhaCMS/piranha.core#1591]
TL;DR: I wanted to figure out how I can switch the layouts for different websites.

**EDIT: I figured out a bit of a workaround using Modules - see my comment reply below.**

I have been unsuccessful in trying to switch the Layout based upon a `SiteType`. I was trying to handle this in the `_ViewStart.cshtml` file.

I tried something like:

    string CurrentSite = Convert.ToString(WebApp.Api.SiteTypes);
    switch (CurrentSite)
    {
        case "Standard Site": // also tried "StandardSite"
            Layout = "~/Pages/Shared/_Layout2.cshtml";
            break;
        default:
            Layout = "~/Pages/Shared/_Layout.cshtml";
            break;
    }
   
but `CurrentSite` is `null`. I presume it's just not picked up at that point; otherwise I'm totally off-base.

I am using the `RazorWeb` project from the Piranha source code.  The Site Type is defined there like this:

    [SiteType(Title = "Standard Site")]
    public class StandardSite : SiteContent<StandardSite>
    {
        [Region]
        public HtmlField Footer { get; set; }
    }

Any thoughts? I was thinking of using a different Site Type for Website1 and Website 2, so that I could use different layouts and sort of mimic having a different theme for each, respectively. I might be going about this totally wrong. I am wondering if I could accomplish the same trying to switch based upon the URL. But as this is on localhost, I'm not using a www.website1.com address. Again, I could be going about this totally wrong.

**Comment by remesq** [comment]
Spending a bit more time with the system, I think I have managed to workaround some of this problem. I just have to keep tweaking it.

I am sure there is room for improvement, so if anyone has suggestions it would be appreciated. :)

I created a Module for site/page/other models that have their own respective ~/Pages folders with the associated `.cshtml` files. This at least separates the pages I'd like to use for individual websites, and allows me to style them how I want. I haven't delved too deeply into other content types, but I was able to get this working with a Site Type and a Page Type. I just have to figure out how to get everything else to work.

**Comment by tidyui** [comment]
Hi there! A couple of things. `WebApp.Api.SiteTypes` is the API-service that will access the meta-data for the available registered Site Content Types, it has nothing to do with the actually selected type on the current site. When executing a page or post your can get the id of the current site from `WebApp.Site.Id`. There's also a helper method to load (if available) the current global site content by calling `WebApp.Site.GetContent<T>()` where `T` is the SiteType.

You can absolutely use different subfolders for different pages where each subfolder has its own `_ViewStart.cshtml` file that specifies the Layout that should be used. This solution depends fully on `ASP.NET` and has nothing to do with Piranha. The downside with it is of course that the same Page can't be used with different layouts.

My suggestion would be to create a `SiteType` which has a region containing a `SelectListField<T>` where `T` should be the type of a custom enum specifying the available themes. Then you would select the theme on site level, and you could load it in the Page and set the Layout, for example:

~~~ csharp
public enum ThemeTypes
{
    Default,
    Fancy
}
~~~

**Comment by remesq** [comment]
Awesome to know. From what I gather, I have multiple options to tackle this problem. Closing.

**Comment by xdmushui** [comment]
> Awesome to know. From what I gather, I have multiple options to tackle this problem. Closing.

How did you solve this problem? Can you share your code?

**Comment by remesq** [comment]
@xdmushui I don't have a project setup on Github to share as I have not gotten around to trying TidyUI's suggestion above to use a `SelectListField<T>` yet. It's likely too complicated for my skillset and I'm researching how to do it. I found it a bit easier to just set up a different project for each websites "theme"

I just create a Razor Class Library to house my models/pages. I set up a Site Type with some Regions to do my Header/Footer content that is the same on every page. I then use different Page Types linked to `.cshtml` pages with their own styles.

I'm not sure if that's what you are after.

**Comment by josip8** [comment]
For anyone wondering (just to reiterate what tidyui said), I created new site type like this:
```
[SiteType(Title = "Test Site")]
public class TestSite : SiteContent<TestSite>
{
    [Region]
    public SelectField<ThemeStyles> ThemeStyle { get; set; }
}


public enum ThemeStyles
{
    [Display(Description = "Theme 1 - Blue")]
    ThemeBlue,
    [Display(Description = "Theme 2 - Red")]
    ThemeRed
}
```
And than in `_Layout.cshtml` you can access this site property like this:
```
@{
  var theme = (await WebApp.Site.GetContentAsync<TestSite>())?.ThemeStyle;
}
```
